src/common/proxy_wasm/ngx_proxy_wasm.c

```diff
@@ -242,7 +242,8 @@ ngx_proxy_wasm_ctx_alloc(ngx_pool_t *pool)
         return NULL;
     }
 
-    pwctx->pool = pool;
+    pwctx->pool = ngx_create_pool(128, pool->log);
```

Comment:
The call can fail, let's catch the error; perhaps also make `128` a `#define`? It probably is a little low too, no?

Reply:
Another note too, I think we could first create the pool, then allocate `pwctx` inside of it? That way no need to keep track of the parent pool at all, and it could even disappear from this function's arguments.

Reply:
Yes, it is how it works. Maybe we can have `NGX_WASM_POOL_SIZE_S` and `NGX_WASM_POOL_SIZE_M` with `128` and `512`? Fine to hard-code `512` as well given we already do it, we can `#define` later.

Reply:
> I think we could first create the pool, then allocate pwctx inside of it?

Can do, if you prefer this pattern. Personally, I always try to organize memory allocations as an ownership tree, that helps me track the lifetimes of allocations by looking at the structure definitions rather than by tracking the execution flow. Having the structure that owns the pool stored inside the pool itself makes the ownership relationship circular, which takes more attentive coding as it gets harder and harder to track all flows in your head as things grow, so it's a pattern I've always avoided. (I believe it's the kind of pattern you cannot even represent (easily?) in Rust, for example — I guess I've been "thinking in Rust" before Rust existed :sweat_smile: )

I wanted to avoid refactors to keep the PR small and the review easier, but if it were up to me, I'd prefer instead to make `ngx_proxy_wasm_ctx_alloc` and `ngx_proxy_wasm_ctx_destroy` more symmetrical not by removing the parent pool argument, but by passing it to the destroy function _as well_, making them more like `ngx_pcalloc` and `ngx_pfree` (which would then remove the `parent_pool` field from the struct).

That, of course, requires that the owner of the parent pool is reachable at destruction time. From a quick look at the code it seems that those alloc and destroy functions are called at pretty different places (one under `http/`, another under `common/`), so a change to make things more symmetrical would be more code surgery than I think should happen in this PR (though I still think that it could have potential benefits, at the very least to force a rethink/clarification of the pool lifetimes). I'm not a big fan of the `parent_pool` field either (I try to avoid "backlinks" to the owner structure whenever possible), but at least I like it better than the apparently-circular ownership. But I can go either way, no strong feelings here, just talking shop to share my thought process.